nav2_planner/src/planner_server.cpp

```diff
@@ -241,26 +241,36 @@ PlannerServer::computePlan()
       return;
     }
 
+    // Changing the start and goal pose frame to the global_frame_ of costmap_ros_ if needed
+    geometry_msgs::msg::PoseStamped goal_pose = goal->goal;
+    if (!costmap_ros_->getPoseInGlobalFrame(start) ||
+        !costmap_ros_->getPoseInGlobalFrame(goal_pose)) {
+      RCLCPP_WARN(
+        get_logger(), "Could not transform the start or goal pose in the costmap frame");
+      action_server_->terminate_current();
+      return;
+    }
+
     if (action_server_->is_preempt_requested()) {
       goal = action_server_->accept_pending_goal();
     }
 
-    result->path = getPlan(start, goal->goal, goal->planner_id);
+    result->path = getPlan(start, goal_pose, goal->planner_id);
 
     if (result->path.poses.size() == 0) {
       RCLCPP_WARN(
         get_logger(), "Planning algorithm %s failed to generate a valid"
         " path to (%.2f, %.2f)", goal->planner_id.c_str(),
-        goal->goal.pose.position.x, goal->goal.pose.position.y);
+        goal_pose.pose.position.x, goal_pose.pose.position.y);
       action_server_->terminate_current();
       return;
     }
 
     RCLCPP_DEBUG(
       get_logger(),
       "Found valid path of size %lu to (%.2f, %.2f)",
-      result->path.poses.size(), goal->goal.pose.position.x,
-      goal->goal.pose.position.y);
+      result->path.poses.size(), goal_pose.pose.position.x,
```

Comment:
Still missing one https://github.com/ros-planning/navigation2/blob/456b6fd4af5c8697d7b37faf47f8535537149a8c/nav2_planner/src/planner_server.cpp#L292-L293

Reply:
No, these lines are outside the block that defines goal_pose